### PR TITLE
fix typo in condition conjunction

### PR DESCRIPTION
**Pull request by ArsSirek**
#### Description of changes: fix typo,
the condition in the [Google Auth example](
https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#google-sign-in-react) where we check if the google with google accounts available on a page should be `||` not '&&' 

```
 useEffect(() => {
  // Check for an existing Google client initialization
    if (!window.google && !window.google?.accounts) createScript();
  }, []);
```
should have been -> 
```
 useEffect(() => {
  // Check for an existing Google client initialization
    if (!window.google || !window.google.accounts) createScript();
  }, []);
  ```

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by cwomack** [comment]
Hello, @ArsSirek.  We appreciate you taking the time to submit this PR. 

The use of the “and” operator (`&&`) ensures that we check to see if both `window.google` and `window.google?accounts` are false, then proceed to execute `createScript()`.  If we change that to use the "or" operator (`||`), then `createScript()` may run if only 1 of them is true.

Also, the removal of the optional chaining operator in `window.google?accounts` can run the risk of a `TypeError` being thrown should `window.google?accounts` or `window.google` be undefined or null.  

Because of the above reasons, I believe the code samples provided in our documentation should be proper.  If you have further questions or rationale on why the suggested changes would improve the behavior of the code, feel free to comment back!

File: src/fragments/lib/auth/js/advanced.mdx [hunk 1]
```diff
@@ -189,7 +189,7 @@ import { Auth } from 'aws-amplify';
 const SignInWithGoogle = () => {
   useEffect(() => {
   // Check for an existing Google client initialization
-    if (!window.google && !window.google?.accounts) createScript();
+    if (!window.google || !window.google.accounts) createScript();
```
Comment (stocaaro) [inline src/fragments/lib/auth/js/advanced.mdx:192]:
```suggestion
    if (!window.google.accounts) createScript();
```

File: src/fragments/lib/auth/js/advanced.mdx [hunk 2]
```diff
@@ -250,7 +250,7 @@ import { Auth } from 'aws-amplify';
...
```
Comment (stocaaro) [inline src/fragments/lib/auth/js/advanced.mdx:253]:
```suggestion
    if (!window.google.accounts) createScript();
```

**Review by stocaaro — COMMENTED** [review]
The optional chaining operator should be enough here, I don't think we need to check `window.google` first.  I would recommend we remove these and merge this fix.  

Thank you for contributing this change. Catches and resolves the corner case where `window.google` exists, but doesn't have `accounts` on it.

**Comment by cwomack** [comment]
@ArsSirek, thank you for taking the time to submit this PR! I've had it reviewed internally with our team and we've made just a few minor suggestions.  Let us know if you agree and we'd be happy to get this reviewed and merged in.

**Comment by ArsSirek** [comment]
Hi! Thank you all for taking time to review and the suggestions!

@cwomack - I updated the commit (force pushed to my fork to not increase the commits amount on this issue, hope it is okey)

There was one more check that I thought should remain in the same manner as this one, so I removed unnecessary `window.google &&` there as well. 

It originally was `window.google && window.google?.accounts` and the condition in this issue should have been a negation of this one, so I updated both now

**Review by cwomack — APPROVED** [review]
LGTM

**Review by katiegoines — APPROVED** [review]

**Pull request merged**